packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -137,6 +137,8 @@ export function PrivateBlockToolbar( {
 				! _isZoomOut &&
 				parentBlockType &&
 				getBlockEditingMode( parentClientId ) !== 'disabled' &&
+				getBlockEditingMode( selectedBlockClientId ) !==
+					'contentOnly' &&
```

Comment:
Nit (non-blocking): the block editing mode for the selected one is already available here. See the `editingMode` variable above.

Reply:
Updated in [8a0be0e](https://github.com/WordPress/gutenberg/pull/67395/commits/8a0be0e77298698fb2ac4c68c44a3b96ffec5cea).